pac4j-core/src/main/java/org/pac4j/core/exception/RequiresHttpAction.java

```diff
@@ -90,6 +90,22 @@ public static RequiresHttpAction unauthorized(final String message, final WebCon
         context.setResponseStatus(HttpConstants.UNAUTHORIZED);
         return new RequiresHttpAction(message, HttpConstants.UNAUTHORIZED);
     }
+
+    /**
+     * Build a digest auth popup credentials.
+     *
+     * @param message message
+     * @param context context
+     * @param realmName realm name
+     * @return a digest auth popup credentials
+     */
+    public static RequiresHttpAction unauthorizedDigest(final String message, final WebContext context, final String realmName, final String qop, final String nonce) {
+        if (CommonHelper.isNotBlank(realmName)) {
```

Comment:
Yes, it looks specific to the digest auth, but at the same time, it fits well here.